app/domain/authentication/strategy.rb

```diff
@@ -77,6 +77,24 @@ def conjur_token(input)
       raise e
     end
 
+    def conjur_token_oidc(input)
+      authenticator = authenticators[input.authenticator_name]
+
+      validate_authenticator_exists(input, authenticator)
+
+      # this method also injects the username to input
```

Comment:
Our new plan involves releasing with this version without separate endpoints.

However, I realized we can (and should) still split up the "get user details" and "check if valid" steps.

I would probably be inclined to create two command objects for these steps, although it could also be achieved by adding a new method to the authenticator.

This will allow us to avoid the hack above that mutates input, which needs to remain immutable.

As a separate issue which I should probably take care of myself, since it affects code unrelated to OIDC itself, but which is worth pointing out, we should eventually have an OIDC specific "Strategy" object rather than editing the existing one.  This is related to this existing issue: https://github.com/cyberark/conjur/issues/747